app/src/main/java/com/squirtles/musicroad/map/MapViewModel.kt

```diff
@@ -143,38 +149,45 @@ class MapViewModel @Inject constructor(
             _centerLatLng.value?.run {
                 val radiusInM = leftTop.distanceTo(this)
                 fetchPickUseCase(this.latitude, this.longitude, radiusInM)
-                    .onSuccess { pickList ->
-                        val newKeyTagMap: MutableMap<MarkerKey, String> = mutableMapOf()
-                        pickList.forEach { pick ->
-                            newKeyTagMap[MarkerKey(pick)] = pick.id
-                            _picks[pick.id] = pick
-                        }
-                        _clickedMarkerState.value.clusterPickList?.let { clusterPickList -> // 클러스터 마커가 선택되어 있는 경우
-                            val updatedPickList = mutableListOf<Pick>()
-                            clusterPickList.forEach { pick ->
-                                _picks[pick.id]?.let { updatedPick ->
-                                    updatedPickList.add(updatedPick)
-                                }
+                    .collect { pickList ->
+                        mutex.withLock {
+                            val newKeyTagMap: MutableMap<MarkerKey, String> = mutableMapOf()
+                            pickList.forEach { pick ->
+                                newKeyTagMap[MarkerKey(pick)] = pick.id
+                                _picks[pick.id] = pick
                             }
-                            _clickedMarkerState.emit(_clickedMarkerState.value.copy(clusterPickList = updatedPickList.toList())) // 최신 픽 정보로 clusterPickList 업데이트
+
+                            // 업데이트된 리스트에 기존 픽이 없으면 삭제된 것이므로 _picks와 clusterer에서 삭제
+                            // 이거 없으면 다른 기기에서 실제로 삭제는 되어 잇는데 지도에 그대로 남아잇음
+                            val deletedKeyList = _picks.keys
+                                .filter { pickId ->
+                                    pickId !in newKeyTagMap.values
+                                }
+                                .mapNotNull { pickId ->
+                                    _picks[pickId]
+                                }
+                                .onEach { pick ->
+                                    _picks.remove(pick.id)
+                                }
+                                .map { pick ->
+                                    MarkerKey(pick)
+                                }
```

Comment:
기존에 있던 pick을 비교해서 지우는 코드를 조금 더 간결화할 수 있을 것 같습니다!

```suggestion
                              val deletedKeyList = _picks.keys
                                .filterNot { it in newKeyTagMap.values }
                                .mapNotNull { pickId ->
                                    _picks.remove(pickId)?.let { MarkerKey(it) }
                                }
```